lib/diagnostic.js

```diff
@@ -1,21 +1,45 @@
 'use strict'
 
+/*
+Comfort: 2
+Clarity: 2
+
+I used prior lessons, Google, and my class notes to help with this diagnostic.
+*/
+
+
+
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
 
+  //for the callbacks, these are called for each element, correct?
+
+  let result = previous
+
+  if (result < current) {
+    return result
+  } else {
+    result = current
+    return result
+  }
 }
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
+
+// i don't understand for sure if previous or current are arrays or values in
+// array; just discovering my questions
```

Comment:
So `previous` is sort of the accumulation of all items up until this point, and `current` is simply the current value.